README.md

```diff
@@ -324,6 +321,11 @@ container.resolve('printTime')()
 container.resolve('printTime')()
 ```
 
+If you want a mismatched configuration like this to error, set
```

Comment:
Based on our discussion, perhaps we should flag this as a highly recommended option to enable.

Reply:
@jeffijoe great point. I'll mark it as recommended in the uses in the README, and also update the examples to use it.